index.bs

```diff
@@ -2916,6 +2916,246 @@ value and terminate the operation.
 
 </div>
 
+### Signal Credential Changes to the Authenticator - PublicKeyCredential's `signal()` Method ### {#sctn-signal}
+
+<xmp class="idl">
+    partial interface PublicKeyCredential {
+        static Promise<undefined> signal(optional PublicKeyCredentialSignalOptions options = {});
+    };
+
+    dictionary PublicKeyCredentialSignalOptions {
+        UnknownCredentialIdReport              unknownCredential;
+        CurrentCredentialsReport               currentCredentials;
+    };
+
+    dictionary UnknownCredentialIdReport {
+        required USVString                     rpId;
+        required Base64URLString               credentialId;
+    };
+
+    dictionary CurrentCredentialsReport {
+        required USVString                     rpId;
+        required Base64URLString               userId;
+        CurrentCredentialUserDetails           user;
+        sequence<Base64URLString>              allAcceptedCredentialIds;
+    };
```

Comment:
It could be argued that this report identifies itself as an **exhaustive** list of **recognized** credential IDs. If we accept that, could we shorten `allAcceptedCredentialIds` to `credentialIds`? Make it look more like `UnknownCredentialIdReport.credentialId` for brevity 🤔 

```suggestion
    dictionary CurrentCredentialsReport {
        required USVString                     rpId;
        required Base64URLString               userId;
        CurrentCredentialUserDetails           user;
        sequence<Base64URLString>              credentialIds;
    };
```

Reply:
Are you suggesting that "Current" in "CurrentCredentialsReport" is enough of an indication that the RP must put all credential ids in here, and that it's superfluous to reinforce that with "allAccepted"?

My initial reaction is to keep it as is, because the consequences can be drastic: A provider may, and likely will delete any credential it has that is not included on the list. I.e. the meaning of the field here is very different from the credentialId field in the UnknownCredentialIdReport.

Reply:
Hmm, we could go a slightly different route here and rename `CurrentCredentialsReport` to `AllAcceptedCredentialsReport`? Paired with a rename of `allAcceptedCredentialIds` to `credentialIds` it might be more explicit 🤔 

To keep the bikeshedding going, there might also be a possibility of leaving it called `CurrentCredentialsReport` but renaming `allAcceptedCredentialIds` to `allCurrentCredentialIds`.

Having said this I think what I'm trying to do is unify the wording here. I don't particularly care if we go with "current" or "accepted", but I do feel as though there should be one word used for consistency.

Reply:
> It could also be a separate parameter (credentialIds and deleteOtherCredentials)

It should not be an explicit list of credentials to be deleted because otherwise the relying party has to keep that list ~forever.

> Hmm, we could go a slightly different route here and rename CurrentCredentialsReport to AllAcceptedCredentialsReport?

How about `AllAcceptedCredentialsReport` and keep `allAcceptedCredentialIds`? That way we get consistency and an explicit string. I'll also add a warning that this will remove or hide all credentials not in the list.

Reply:
I added a note warning of deletion. Splitting the username / displayname updates should help avoid the deletion footgun when a developer is only interested in updating that.

The way to invoke the API now is:

```javascript
PublicKeyCredential.signalAllAcceptedCredentialIds({
    rpId: "example.com",
    userId: "aabbcc",  // user handle, base64url.
    allAcceptedCredentialIds: [
        "bb",
    ]
});
```

I cannot think of something more obvious short of `credentialsNotPresentInTheFollowingListShouldBeDeletedIKnowWhatImDoing`.

Reply:
> How about `AllAcceptedCredentialsReport` and keep `allAcceptedCredentialIds`? That way we get consistency and an explicit string. I'll also add a warning that this will remove or hide all credentials not in the list.

I think this is probably the way to go. An issue with potential names like `RemoveUnlistedCredentialIds` and `DeleteNonListedCredentialIds` could be that different providers do different things. As suggested earlier, maybe a provider wants to tag a not in the list passkey vs delete it outright. "Accepted" vs "Remove" or "Delete" gives providers greater flexibility to handle such a signal without making it confusing to explain to RP's the impact of making such a call.